test/test_string_map.cpp

```diff
@@ -21,6 +21,38 @@
 #include "rcutils/error_handling.h"
 #include "rcutils/types/string_map.h"
 
+class TestStringMap : public ::testing::Test
+{
+protected:
+  void SetUp() final
+  {
+    // Reset rcutil error global state in case a previously
+    // running test has failed.
+    rcutils_reset_error();
+
+    allocator = rcutils_get_default_allocator();
+    failing_allocator = get_failing_allocator();
+    string_map = rcutils_get_zero_initialized_string_map();
+  }
+
+  void TearDown() final
+  {
+    // Ensure the next call will be successful even if the test
+    // failed.
+    rcutils_reset_error();
+
+    // In some cases, the test may exit before the list has been initialized.
+    // This is OK to ignore.
+    const rcutils_ret_t ret = rcutils_string_map_fini(&string_map);
+    EXPECT_TRUE(ret == RCUTILS_RET_OK || ret == RCUTILS_RET_NOT_INITIALIZED);
```

Comment:
I really don't like this. I think calls to `init` and `fini` should be matched. I see that they are matched in the `getters_initialize_to_zero` test case below, so is this even needed? If a test case is missing a `fini` and therefore this case is needed, then I think it's a bug in the test case.

To a lesser degree I dislike the defensive `rcutils_reset_error` calls, as they lead to cases which I'll point out below.

I think this would be better with the entire `TearDown` removed.

Reply:
> so is this even needed?

Nevermind, I see that `getters_size_null_size` init's but does not fini, which is inconsistent with `getters_initialize_to_zero`. Is there a reason for that? I still think removing this from the tear down and doing the fini in the test case is the best solution.

Reply:
Removed the TearDown.